### DISCUSSION TITLE
How do I combine my `AssetSelection` as unions, intersections, or differences?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18406] [Q&A] [answered]
Hello, I am trying to pass multiple dbt-based AssetSelection resources to a job definition. It doesn't work. Is there a better way to do this? Example:

```python
remit_professional_pipeline = define_asset_job(
    name="Remit_Professional_Build",
    selection=[
        remit_professional_asset_selector,
        remit_professional_end_table_selector
    ],
)
```

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C04CW71AGBW/p1701364777305209?thread_ts=1701364777.305209&cid=C04CW71AGBW).*

**Comment by rexledesma** [comment, marked as answer]
You can combine your `AssetSelection` objects using the “|”, “&”, and “-” operators to create unions, intersections, and differences of selections, respectively.

For example, to select the union of your two selection objects:

```python
remit_professional_pipeline = define_asset_job(
    name="Remit_Professional_Build",
    selection=(
        remit_professional_asset_selector | remit_professional_end_table_selector
    ),
)
```

References:
- https://docs.dagster.io/_apidocs/assets#dagster.AssetSelection
